addons/project/models/project_task.py

```diff
@@ -526,7 +526,7 @@ def _compute_subtask_count(self):
 
     @api.onchange('company_id')
     def _onchange_task_company(self):
-        if self.project_id.company_id and self.project_id.company_id != self.company_id:
+        if self.project_id.company_id and self.company_id and self.project_id.company_id != self.company_id:
```

Comment:
I think the problem is because the company_id is not correctly set by default.

Can you try like that: 

```suggestion
diff --git a/addons/project/models/project_task.py b/addons/project/models/project_task.py
index d45445207d12..eef17fed79cf 100644
--- a/addons/project/models/project_task.py
+++ b/addons/project/models/project_task.py
@@ -946,6 +946,8 @@ class Task(models.Model):
             project = self.env['project.project'].browse(project_id)
             if project.analytic_account_id:
                 vals['analytic_account_id'] = project.analytic_account_id.id
+            if 'company_id' in default_fields and 'default_project_id' not in self.env.context:
+                vals['company_id'] = project.sudo().company_id
         elif 'default_user_ids' not in self.env.context and 'user_ids' in default_fields:
             user_ids = vals.get('user_ids', [])
             user_ids.append(Command.link(self.env.user.id))
```

Normally with that suggestion, we should no longer need to alter `_onchange_task_company` 🙂 

ps: I haven't tested my suggestion